UefiPayloadPkg/Library/FdtParserLib/FdtParserLib.c

```diff
@@ -0,0 +1,625 @@
+  UINT32                                *Data32;
+  UINT64                                *Data64;
+  UINT64                                StartAddress;
+  INT32                                 SubNode;
+  UINT64                                NumberOfBytes;
+  UINT32                                Attribute;
+  UINT8                                 ECCAttribute;
+  UINT32                                ECCData, ECCData2;
+  UNIVERSAL_PAYLOAD_SERIAL_PORT_INFO    *Serial;
+  UNIVERSAL_PAYLOAD_PCI_ROOT_BRIDGES    *PciRootBridgeInfo;
+  UNIVERSAL_PAYLOAD_ACPI_TABLE          *PlatformAcpiTable;
+  UNIVERSAL_PAYLOAD_SMBIOS_TABLE        *SmbiosTable;
+  UEFI_PAYLOAD_DEBUG_PRINT_ERROR_LEVEL  *DebugPrintErrorLevelInfo;
+  UNIVERSAL_PAYLOAD_BASE                *PayloadBase;
+  EFI_PEI_GRAPHICS_INFO_HOB             *GraphicsInfo;
+  EFI_PEI_GRAPHICS_DEVICE_INFO_HOB      *GraphicsDev;
+  // UINT8                                 SizeOfMemorySpace;
+  UINT64                FrameBufferBase;
+  UINT64                FrameBufferSize;
+  UINTN                 MinimalNeededSize;
+  EFI_PHYSICAL_ADDRESS  FreeMemoryBottom;
+  EFI_PHYSICAL_ADDRESS  FreeMemoryTop;
+  EFI_PHYSICAL_ADDRESS  MemoryBottom;
+  EFI_PHYSICAL_ADDRESS  MemoryTop;
+  BOOLEAN               IsHobContructed;
+  UINTN                 NewHobList;
+
+  Fdt               = FdtBase;
+  Depth             = 0;
+  FrameBufferBase   = 0;
+  FrameBufferSize   = 0;
+  MinimalNeededSize = FixedPcdGet32 (PcdSystemMemoryUefiRegionSize);
+  IsHobContructed   = FALSE;
+  NewHobList        = 0;
+  PlatformAcpiTable = NULL;
+  SmbiosTable       = NULL;
+
+  DEBUG ((DEBUG_INFO, "FDT = 0x%x  %x\n", Fdt, Fdt32ToCpu (*((UINT32 *)Fdt))));
+  DEBUG ((DEBUG_INFO, "Start parsing DTB data\n"));
+  DEBUG ((DEBUG_INFO, "MinimalNeededSize :%x\n", MinimalNeededSize));
+
+  for (Node = FdtNextNode (Fdt, 0, &Depth); Node >= 0; Node = FdtNextNode (Fdt, Node, &Depth)) {
+    if (Depth != 1) {
+      continue;
+    }
+
+    NodePtr = (FDT_NODE_HEADER *)((CONST CHAR8 *)Fdt + Node + Fdt32ToCpu (((FDT_HEADER *)Fdt)->OffsetDtStruct));
+    DEBUG ((DEBUG_INFO, "\n   Node(%08x)  %a   Depth %x", Node, NodePtr->Name, Depth));
+    // memory node
+    if (AsciiStrnCmp (NodePtr->Name, "memory@", AsciiStrLen ("memory@")) == 0) {
+      for (Property = FdtFirstPropertyOffset (Fdt, Node); Property >= 0; Property = FdtNextPropertyOffset (Fdt, Property)) {
+        PropertyPtr = FdtGetPropertyByOffset (Fdt, Property, &TempLen);
+        TempStr     = FdtGetString (Fdt, Fdt32ToCpu (PropertyPtr->NameOffset), NULL);
+        if (AsciiStrCmp (TempStr, "reg") == 0) {
+          Data64        = (UINT64 *)(PropertyPtr->Data);
+          StartAddress  = Fdt64ToCpu (*Data64);
+          NumberOfBytes = Fdt64ToCpu (*(Data64 + 1));
+          DEBUG ((DEBUG_INFO, "\n         Property(%08X)  %a", Property, TempStr));
+          DEBUG ((DEBUG_INFO, "  %016lX  %016lX", StartAddress, NumberOfBytes));
+          if (IsHobContructed != TRUE ) {

```

Comment:
Is it a typo for IsHobContructed (change to IsHobConstructed)? and you don't need compare it with TRUE since it is BOOLEAN type.